aptos-move/block-executor/src/counters.rs

```diff
@@ -22,12 +24,52 @@ pub static SPECULATIVE_ABORT_COUNT: Lazy<IntCounter> = Lazy::new(|| {
     .unwrap()
 });
 
-/// Count of times a transaction got suspended due to an estimated r/w dependency.
-pub static DEPENDENCY_SUSPEND_COUNT: Lazy<IntCounter> = Lazy::new(|| {
-    register_int_counter!(
-        "aptos_execution_dependency_suspend_count",
-        "Count write estimates encountered when reading in parallel execution \
-        (typically leading to a suspension / waiting)"
+pub static VM_INIT_SECONDS: Lazy<Histogram> = Lazy::new(|| {
+    register_histogram!(
+        // metric name
+        "aptos_execution_vm_init_seconds",
+        // metric description
+        "The time spent in seconds in initializing the VM in the block executor",
+        exponential_buckets(/*start=*/ 1e-3, /*factor=*/ 2.0, /*count=*/ 20).unwrap(),
+    )
+    .unwrap()
+});
+
+pub static TASK_VALIDATE_SECONDS: Lazy<Histogram> = Lazy::new(|| {
+    register_histogram!(
+        // metric name
+        "aptos_execution_task_validate_seconds",
+        // metric description
+        "The time spent in task validation in Block STM",
+        exponential_buckets(/*start=*/ 1e-3, /*factor=*/ 2.0, /*count=*/ 20).unwrap(),
+    )
+    .unwrap()
+});
+
+pub static TASK_EXECUTE_SECONDS: Lazy<Histogram> = Lazy::new(|| {
+    register_histogram!(
+        // metric name
+        "aptos_execution_task_execute_seconds",
+        // metric description
+        "The time spent in seconds for task execution in Block STM",
+        exponential_buckets(/*start=*/ 1e-3, /*factor=*/ 2.0, /*count=*/ 20).unwrap(),
+    )
+    .unwrap()
+});
+
+pub static DEPENDENCY_WAIT_SECOND: Lazy<Histogram> = Lazy::new(|| {
```

Comment:
*_SECONDS to be consistent with the other names?

by the way why seconds and not MILLIS everywhere? Isn't the number usually in the milliseconds? Do we use seconds everywhere is that why?

Reply:
>by the way why seconds and not MILLIS everywhere? Isn't the number usually in the milliseconds? Do we use seconds everywhere is that why?

We use seconds everywhere, this is because of the histogram API, uses `elapsed_sec()` API to measure the time.